source/common/http/conn_manager_impl.h

```diff
@@ -402,12 +420,43 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
         }
       }
     }
-    // Indicates which filter to start the iteration with.
-    enum class FilterIterationStartState { AlwaysStartFromNext, CanStartFromCurrent };
+
+    void decodeHeaders(RequestHeaderMap& headers, bool end_stream) {
+      decodeHeaders(nullptr, headers, end_stream);
+    }
+    void decodeData(Buffer::Instance& data, bool end_stream) {
+      decodeData(nullptr, data, end_stream, FilterIterationStartState::CanStartFromCurrent);
+    }
+    void decodeTrailers(RequestTrailerMap& trailers) { decodeTrailers(nullptr, trailers); }
+    void decodeMetadata(MetadataMap& metadata_map) { decodeMetadata(nullptr, metadata_map); }
 
     void addStreamDecoderFilterWorker(StreamDecoderFilterSharedPtr filter, bool dual_filter);
     void addStreamEncoderFilterWorker(StreamEncoderFilterSharedPtr filter, bool dual_filter);
 
+    void disarmRequestTimeout();
+    void maybeEndDecode(bool end_stream);
```

Comment:
again if we can improve comments as we do this refactor, lets.
TODO OK if there's more refactor coming here.